src/main/java/lotto/domain/entity/Lotto.java

```diff
@@ -0,0 +1,63 @@
+package lotto.domain.entity;
+
+import static lotto.configuration.IntegerConstants.MAX_NUMBER_COUNT_OF_LOTTO;
+import static lotto.configuration.IntegerConstants.MAX_NUMBER_RANGE;
+import static lotto.configuration.IntegerConstants.MIN_NUMBER_RANGE;
+import static lotto.configuration.StringConstants.SEPERATE_STANDARD;
+
+import camp.nextstep.edu.missionutils.Randoms;
+import java.util.Collections;
+import java.util.List;
+import lotto.util.IntegerListUtil;
+import lotto.util.StringUtil;
+import lotto.validation.IntegerListValidator;
+
+public class Lotto {
+    private final List<Integer> numbers;
+
+    public Lotto(List<Integer> numbers) {
+        validate(numbers);
+        this.numbers = numbers;
+    }
+
+    public static Lotto create(String input) {
+        input = StringUtil.deleteAllSpaces(input);
+        List<Integer> numbers = IntegerListUtil.parseListseperatedBy(
+                input, SEPERATE_STANDARD.getValue());
+        return new Lotto(numbers);
+    }
```

Comment:
생성자와 정적으로 생성하는 구현을 같이 두신 이유가 있을까요? 
정적 팩토리 방식과 생성자 방식의 구현을 같은 객체에 두는 방법이 가지는 장점이 있다면 알려주실수있을까요?

```suggestion
    public Lotto(String input) {
        input = StringUtil.deleteAllSpaces(input);
        List<Integer> numbers = IntegerListUtil.parseListseperatedBy(
                input, SEPERATE_STANDARD.getValue());
        this.numbers = numbers;
    }
```

Reply:
위에분께서 말씀하신대로 정적 팩토리 메서드를 사용하는 이유가 있으실까요?? 생성자를 이용해도 충분할 것 같은데 다른 이유가 궁금합니다!

Reply:
정적 팩터리 메서드는 **이펙티브 자바**라는 도서에서 읽은 내용인데, 
저는 개인적으로 생성자보단 정적 팩토리 메서드를 통해 객체를 생성하는 것을 선호합니다!
추후 확장을 고려한다면 훨씬 더 이점이 클 것이라 생각해요!

- 메소드 이름을 가질 수 있습니다. create라는 이름은 객체를 생성한다는 의미로 바로 파악될 수 있습니다.
- 생성자와 다르게, 호출한다고 해서 매번 꼭 해당 클래스의 인스턴스가 생성되는 것이 아닙니다. 조건부로 객체를 반환해줄 수도 있습니다.
- 반환 타입의 활용성이 다양해집니다. 반환 타입에 인터페이스를 적을 수 있으며, 반환값은 반환 타입의 자식일 수도 있습니다. 입력값에 따라 다른 자식 타입을 반환해줄 수도 있습니다.

Lotto 객체를 생성하는 방법에는 여러가지가 있을 수 있다고 생각합니다.
바로 아래 👇에도 createRandomSorted() 라는 이름으로, 정렬된 랜덤 Lotto 객체를 생성하는 정적 팩터리 메서드를 사용하고 있습니다!

가장 핵심은, 결국 **`String input`을 List<Integer>로 바꿔줘야 하는데, `public Lotto(String input)`을 둔다면 `validate(numbers);`라는 코드가 중복됩니다.**
 
~~~
    public Lotto(String input){
        //String input만을 위한 검증 로직들
        return new Lotto(List<<Integer>()); //생성자에서는 다른 생성자 호출 불가능
    }
~~~
이런 코드는 불가능하기에,
~~~
    public Lotto(List<Integer> numbers) {
        validate(numbers); //중복
        this.numbers = numbers;
    }

    public Lotto(String input){
        //String input만을 위한 검증 로직들
        validate(numbers); //중복
        this.numbers = 변환된numbers;
    }
~~~
위와 같이 코드가 중복됩니다.

` public static Lotto create(String input)`은 생성자 `public Lotto(List<Integer> numbers)`를 호출할 수 있습니다.
결국 Lotto 객체를 만들기 위해 필수적이고 공통적인 부분을 생성자에 두고, create 정적 팩터리 메서드에서는 String input만을 위한 검증 로직들만을 담아줬습니다.
String input이 Lotto에 담길 수 있는 숫자인지 검증하는 것이 Lotto 객체의 책임이라 생각하기 때문에, 검증을 내부에서 하려면 정적 팩터리 메서드가 더 유리하다고 판단했습니다!

또, 다른 코드들에서는 정적 팩터리 메서드를 쓸 때 생성자를 private로 두었는데, 여기서 Lotto 객체는 우테코에서 제공해준 테스트에 의해 꼭 필요한지라 public으로 두었네요 😅 만약 제한이 없었다면 아래와 같이 코드를 두었을 것입니다.
1) List<Integer> 만 입력으로 주어졌을때
2) String input만 입력으로 주어졌을 때
3) 주어진 값 없이 랜덤, 정렬된 로또를 생성해야할 때 
4) 모든 Lotto 객체가 반드시 검증하거나 변환해야할 것 (생성자)
이렇게 각각의 경우에서 역할을 분리하고, 확장에도 더욱 유리하다 생각합니다!
~~~
public class Lotto {
    private final List<Integer> numbers;

    private Lotto(List<Integer> numbers) {
        validate(numbers);
        this.numbers = numbers;
    }
    
    public static Lotto create(List<Integer> numbers) {
        return new Lotto(numbers);
    }
    
    public static Lotto create(String input) {
        input = StringUtil.deleteAllSpaces(input);
        List<Integer> numbers = IntegerListUtil.parseListseperatedBy(
                input, SEPERATE_STANDARD.getValue());
        return new Lotto(numbers);
    }

    public static Lotto createRandomSorted() {
        List<Integer> numbers = Randoms.pickUniqueNumbersInRange(
                1, 45, 6);

        return new Lotto(numbers);
    }
~~~